File: jdi framework/jdi-uitest-core/src/main/java/com/epam/jdi/uitests/core/interfaces/base/IElement.java
package com.epam.jdi.uitests.core.interfaces.base;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */

import com.epam.jdi.uitests.core.annotations.JDIAction;

/**
 * Created by Roman_Iovlev on 6/10/2015.
 */
public interface IElement extends IBaseElement, IVisible {

    /**
     * Get element attribute
     *
     * @param name Specify name for attribute
     * @return Returns chosen attribute
     */
    @JDIAction
    String getAttribute(String name);

    /**
     * @param name  Specify attribute name
     * @param value Specify attribute value
     * Waits while attribute gets expected value. Return false if this not happens
     */
    @JDIAction
    void waitAttribute(String name, String value);

    /**
     * @param attributeName Specify attribute name
     * @param value         Specify attribute value
     *                      Sets attribute value for Element
     */
    @JDIAction
    void setAttribute(String attributeName, String value);

}

File: C#.Net/JDI/Core/UnitTestProject1/Interfaces/Base/IElement.cs
namespace JDI_Core.Interfaces.Base
{
    public interface IElement : IBaseElement, IVisible
    {
        /**
    * Get element attribute
    *
    * param 'name' - Specify name for attribute
    * return - Returns chosen attribute
    */
        //TODO:[JDIAction]
        string GetAttribute(string name);

        /**
         * param 'name' - Specify attribute name
         * param 'value' - Specify attribute value
         * Waits while attribute gets expected value. Return false if this not happens
         */
        //TODO:[JDIAction]
        void WaitAttribute(string name, string value);

        /**
         * param 'attributeName' - Specify attribute name
         * param 'value'     -    Specify attribute value
         *                      Sets attribute value for WebElement
         */
        //TODO:[JDIAction]
        void SetAttribute(string attributeName, string value);
    }
}
